Userland/Applications/ImageViewer/ViewWidget.cpp

```diff
@@ -174,24 +217,39 @@ void ViewWidget::open_file(String const& path, Core::File& file)
 
 ErrorOr<void> ViewWidget::try_open_file(String const& path, Core::File& file)
 {
-    // Spawn a new ImageDecoder service process and connect to it.
-    auto client = TRY(ImageDecoderClient::Client::try_create());
-    auto mime_type = Core::guess_mime_type_based_on_filename(path);
-    auto decoded_image_or_none = client->decode_image(TRY(file.read_until_eof()), mime_type);
-    if (!decoded_image_or_none.has_value()) {
+    // FIXME: Figure out an out-of-process decode that works for both vector/raster images.
+    auto file_data = TRY(file.read_until_eof());
```

Comment:
OOP decoding is a really nice feature that IMHO should be more embraced by the project, I would be really sad to see it gone.
Vector graphic support is awesome, but AFAIC it shouldn't come with this cost.

Reply:
There's not really a practical way to do OOP decoding for vector graphics, unlike a bitmap it's not something that can trivially encoded/decoded over IPC. The structure/paths could depend on the format so there's no generic way to send it, it has to be decoded in-process.

Reply:
You could open the decoder (which won't read more than the header), check if it's a vector, if so load the it, otherwise do the loading via the `ImageDecoderServer`?

Reply:
> There's not really a practical way to do OOP decoding for vector graphics

That's not a reason to lose OOP decoding for raster images.

But actually, I don't see what prevents you to pass a `VectorImageFrameDescriptor` through IPC? The pointer hierarchy will be painful, but the actual data boils down to a simple `Vector<DrawCommand>` so that should be doable.

Reply:
There's no format independent way to encode the data, for an raster image it's always a bitmap for a vector format the way the data is structured depends entirely on the format. Encoding it to send over IPC is pretty much just re-creating the file format again.

Reply:
I think reading the header and doing in-process for vector and out of process for raster is a fair compromise.

Reply:
> Encoding it to send over IPC is pretty much just re-creating the file format again.

I didn't see it that way, it's surely a sad thing, but (I guess) at least the dangerous file parsing is done OOP, which is still a major pro.

> I think reading the header and doing in-process for vector and out of process for raster is a fair compromise.

Good enough for me!